Edit Mode

vicr123 opened the pull request:
Implement UI and logic for edit mode

vicr123 committed:
Implement HTTP requests

vicr123 committed:
Add API for authentication token

vicr123 committed:
Initial edit UI

vicr123 committed:
Add extra buttons

Vrabbers committed:
Merge branch 'main' into features/http-requests

Vrabbers committed:
use interactions for auth token

Vrabbers committed:
Woops

vicr123 committed:
Add handling for 401s

vicr123 committed:
Merge branch 'features/http-requests' into features/edit

vicr123 committed:
Create base type for visible map items

vicr123 committed:
Fix syntax

Vrabbers committed:
Merge branch 'main' into features/edit

Vrabbers committed:
woops

Vrabbers committed:
fix null warnings, formatting

Vrabbers commented on BnbnavNetClient/Views/MapView.axaml.cs:
```diff
@@ -90,25 +104,77 @@ protected override void OnInitialized()
+                pos.Y - NodeSize / 2,
+                NodeSize, NodeSize);
+            return (rect, node);
+        }).Where(node => Bounds.Intersects(node.rect)).ToList().AsReadOnly();
```
Please note that there are performance implications to using LINQ and, if my memory serves me right, read-only collections too. Maybe we should reconsider and/or rewrite

vicr123 replied:
@reflectronic your thoughts?

vicr123 replied:
Performance seems to be okay so far - if required we can come back and look at this but I don't think it's an issue right now

Vrabbers committed:
woops x2

Vrabbers commented:
There's a bug.... switching to night bode and then trying to drag crashes... doesn't happen in ``main``, which is pretty werid

Vrabbers committed:
it's fixed ig?

Vrabbers commented:
Seems to be fixed but landmarks still disappear when theme switches

vicr123 committed:
Update drawn items when the window is resized

Vrabbers committed:
new edge flyout mockup

vicr123 committed:
Implement joining node UI

Vrabbers committed:
further fix bugs with bounds and rendering

* improve resize performance

Vrabbers committed:
highlight nodes which are selected

*make brush gray, not black so can be seen over black roads
*load ghost brush from resources
*don't load node resources for each node

Vrabbers committed:
actually fix the bounds issue

Vrabbers committed:
change colors

Vrabbers committed:
edge hit testing

vicr123 committed:
Add console logging for edge

vicr123 committed:
Merge branch 'main' into features/edit

Vrabbers committed:
use built-in dot product

Vrabbers committed:
inertial panning

Vrabbers committed:
Revert "inertial panning"

This reverts commit af971aedd3db4e7e73cc532f78ca122bf3564d2b.

vicr123 committed:
Remove bidi join feature

Vrabbers committed:
Merge branch 'main' into features/edit

Vrabbers committed:
some of the flyout and context menu things

vicr123 committed:
Show join dialog when edges are joined

vicr123 committed:
Introduce edit controllers

vicr123 committed:
Implement more edit controllers

vicr123 committed:
Create extended line class

vicr123 committed:
Implement node moving

vicr123 committed:
Implement Splice UI

vicr123 committed:
Implement Splicing

Vrabbers committed:
fix icon button height

vicr123 committed:
Implement road joining with existing road

vicr123 committed:
Implement creating new roads along with edges

vicr123 committed:
Convert models to classes

vicr123 committed:
Implement node and edge deletion

vicr123 committed:
Update roads correctly

Vrabbers committed:
code cleanup

Vrabbers committed:
clean extended line up

vicr123 committed:
Use proper exceptions for network operations
